### DISCUSSION TITLE
Ability for Cloudinary loader to automatically populate blurDataUrl

### DISCUSSION THREAD

**Discussion by osdiab** [vercel/next.js#30532] [Ideas]
### Describe the feature you'd like to request

Cloudinary has the ability to generate thumbnail-sized versions of images on its end, as do many image CDNs. It would be cool if the `Image` component/`loader` prop was smart enough to be able to generate the corresponding `blurDataUrl` value when `placeholder=blur` using Cloudinary transforms instead of needing to implement that myself.	

### Describe the solution you'd like

If `placeholder` is set to `blur` and the image is using cloudinary or some other custom loader that supports scaling images automatically, use that to generate a low-res thumbnail automatically and insert the value into `blurDataUrl`.

### Describe alternatives you've considered

I'm just implementing this myself and making a wrapper around `Image` at the moment to do it.

**Comment by Perni1984** [comment]
@osdiab could you share your wrapper implementation?

**Reply by osdiab** [reply]
EDIT: i didn't read the context on my own question lol. The below code doesn't set the blur data url.

Sure thing, this is the whole thing (in TypeScript). We handle SVGs vs other images separately so that vectors remain vector images when they're served, but if you don't care about that you can probably simplify this.

(Also FYI svg's sometimes get rendered grainily in Chrome, which you can get around by applying the CSS rules `will-change: transform; transform: scale(.99);` to SVG images (that's our hack at least, idk if there's something better), but that's a separate topic lol.)

```
import type { ImageLoader, ImageProps } from "next/image";
import urlJoin from "url-join";
import Image from "next/image";

// we don't want to apply transforms to SVGs so they get delivered as vectors
export type ImageType = "svg" | "raster";

export const cloudinaryBaseUrl =
  "https://res.cloudinary.com/<your cloudinary cloud name>/image/upload";
export const cloudinaryLoader: ImageLoader = ({ src, width, quality }) =>
  urlJoin(
    cloudinaryBaseUrl,
    ["f_auto", width && `w_${width}`, quality && `q_${quality}`]
      .filter(Boolean)
      .join(","),
    src
  );

export interface CloudinaryImageProps
  extends Omit<ImageProps, "src" | "placeholder" | "loader"> {
  cloudinaryId: string;
}

/**
 * Given a cloudinary ID that may be a vector image (SVG) or a rasterized image
 * (PNG, JPG, etc), displays a scaled version of the image.
 */
export function CloudinaryImage({
  cloudinaryId,
  imageType,
  alt,
  ...rest
}: CloudinaryImageProps): JSX.Element {
  switch (imageType) {
    case "svg":
      return (
        // eslint-disable-next-line @next/next/no-img-element
        <img
          alt={alt}
          {...rest}
          src={urlJoin(cloudinaryBaseUrl, cloudinaryId)}
        />
      );
    case "raster":
      return (
        <Image
          {...rest}
          alt={alt}
          loader={cloudinaryLoader}
          src={cloudinaryId}
        />
      );
  }
}

```

**Reply by benkingcode** [reply]
What does grainily mean?

**Reply by Perni1984** [reply]
@osdiab Maybe I don't get it but in the above Wrapper I don't see where you set the blurDataUrl prop for next/image. I thought that this is the whole idea of the wrapper.

**Reply by osdiab** [reply]
oh whoops i never implemented it lol ignore this answer, it doesn't do it. forgot the context for this particular discussion.

**Reply by osdiab** [reply]
> What does grainily mean?

just that SVGs end up looking ugly due to Chrome rendering artifacts.
